modules/TransformationPlugin/src/main/java/org/gephi/transformation/plugin/TransformationControllerImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.gephi.transformation.plugin;
+
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.GraphController;
+import org.gephi.graph.api.GraphModel;
+import org.gephi.transformation.api.TransformationController;
+import org.gephi.transformation.plugin.operations.MirrorXAxis;
+import org.gephi.transformation.plugin.operations.MirrorYAxis;
+import org.gephi.transformation.plugin.operations.Rotation;
+import org.gephi.transformation.plugin.operations.Scale;
+import org.openide.util.Lookup;
+import org.openide.util.lookup.ServiceProvider;
+
+
+@ServiceProvider(service = TransformationController.class)
+public class TransformationControllerImpl implements TransformationController {
```

Comment:
On one hand you define a SPI, which is nice because one could write a plugin for new transformation. On the other hand, you hard code what transformations can be executed in the controller. Maybe worth rethinking your approach here.